odoo/orm/fields_properties.py

```diff
@@ -341,11 +344,18 @@ def _add_default_values(self, env, values):
 
         return properties_list_values
 
-    def _get_properties_definition(self, record):
-        """Return the properties definition of the given record."""
-        container = record[self.definition_record]
-        if container:
-            return container.sudo()[self.definition_record_field]
+    def _get_properties_definition(self, records):
+        """Return the properties definition of the given records."""
+        if not records:
+            return []
+        first_record = records[0]
+        first_container = first_record[self.definition_record]
+        first_definition = first_container.sudo()[self.definition_record_field]
+        for record in records[1:]:
+            container = record[self.definition_record]
+            if container != first_container and container.sudo()[self.definition_record_field] != first_definition:
+                raise ValidationError(_("records %r don't have the same definition", records))
```

Comment:
we can simplify a bit (2 different definition record will never have the same definition, name are random)
```python
first_record = records[0]
bad_records = records.filtered(lambda r: r[self.definition_record] != first_record[self.definition_record])
if bad_records:
    raise ValidationError(_("records %r don't have the same definition", records))
```